core/computed/metrics/cumulative-layout-shift.js

```diff
@@ -108,13 +112,38 @@ class CumulativeLayoutShift {
    */
   static async compute_(trace, context) {
     const processedTrace = await ProcessedTrace.request(trace, context);
-
     const allFrameShiftEvents =
         CumulativeLayoutShift.getLayoutShiftEvents(processedTrace);
     const mainFrameShiftEvents = allFrameShiftEvents.filter(e => e.isMainFrame);
 
+    let cumulativeLayoutShift;
+    try {
+      // Experimental usage of CDT's trace processor.
+      const processor = new SDK.TraceProcessor.TraceProcessor({
+        LayoutShifts: SDK.TraceHandlers.LayoutShiftsHandler,
+      });
+      // We really want the logic that getLayoutShiftEvents provides for filtering out
+      // some events based on recent input.
+      // Would it make sense to upstream this to CDT?
+      const filteredTrace = processedTrace.frameTreeEvents.filter(event => {
+        if (event.name !== 'LayoutShift') {
+          return true;
+        }
+
+        return allFrameShiftEvents.some(lse => lse.event === event);
+      });
+      await processor.parse(filteredTrace);
+      const data = /** @type {import('../../lib/cdt/SDK.js').TraceProcessorResult} */ (
+        processor.data);
+      cumulativeLayoutShift = data.LayoutShifts.sessionMaxScore;
+    } catch (e) {
+      // Something failed, so fallback to our own implementation.
+      log.error('Error running SDK.TraceProcessor', e);
```

Comment:
I think this deserves to be captured in sentry. If there are situations where the DT trace processor breaks then we should investigate and fix them.